service/history/handler.go

```diff
@@ -550,6 +550,19 @@ func (h *Handler) RespondDecisionTaskFailed(
 		token.RunID,
 		token.ScheduleID))
 
+	if failedRequest != nil && failedRequest.GetCause() == gen.DecisionTaskFailedCauseUnhandledDecision {
```

Comment:
Can you add a comment here which basically says we need a better way to determine if decision task failed due to non-deterministic error?